tests/src/test/scala/akka/kafka/scaladsl/CommittingSpec.scala

```diff
@@ -435,24 +435,32 @@ class CommittingSpec extends SpecBase with TestcontainersKafkaLike with Inside {
     }
 
     // Illustration of https://github.com/akka/alpakka-kafka/issues/942
-    "merge multiple consumers` offsets for different partitions (#942)" in assertAllStagesStopped {
+    "merge multiple consumers' offsets for different partitions (#942)" in assertAllStagesStopped {
       val topic = createTopic(0, partitions = 2)
       val group = createGroupId()
 
       awaitProduce(produceTwoPartitions(topic))
       val consumerSettings = consumerDefaults.withGroupId(group)
       val result =
         Consumer
-          .committableSource(consumerSettings, Subscriptions.assignment(new TopicPartition(topic, partition0)))
+          .committableSource(
+            consumerSettings
+              .withStopTimeout(100.millis), // this consumer actor needs to stay around to receive the commit
+            Subscriptions.assignment(new TopicPartition(topic, partition0))
+          )
           .take(10)
+          // triggers commit timeout as the actor is terminated
+          .delay(50.millis)
```

Comment:
It's good that this resolves the issue, but it looks like it could still arise on systems with slow clocks?  We should try repeating this test (i.e. `sbt test thistest -- -DtimesToRepeat=100`) to see if it fails occasionally.

More long term we should find a way around the stop timeout for happy-path shutdown scenarios.

Reply:
This PR fixes the test, not the issue (and added more information to the commit timeout error).
I'm not sure there is a way to fix this, for how long after emitting data should a source live to receive commits? Should the actor stay around, but the stage complete?